0022-circuit-block.md

```diff
@@ -0,0 +1,364 @@
+
+circuit = QuantumCircuit(6)
+
+# define a twirled layer of CZs
+with circuit.box([PauliTwirl(<specifiers>)]):
+    # both of these single-qubit gates will be composed with the random 
+    # Pauli selections
+    circuit.sx(0)
+    circuit.rz(Parameter("p"), 0)
+    
+    circuit.cz(0, 1)
+    circuit.cz(2, 3)
+    circuit.noop(5)
+
+# define a twirled layer of measurements
+with circuit.box([PauliTwirl(<specifiers>)]):
+    circuit.measure_all()
+```
+
+We envision two main workflows that users will follow once `box`es and annotations become available to specify twirling, one for regular users and another one for power users:
+
+1. The workflow for regular users:
+    - Users initialize a `QuantumCircuit` without boxes, as they do today.
+    - They construct a pass manager that contains a late-stage twirling-decorator pass.
+    - The early passes of the pass manager do steps like routing and layout as usual.
+    - The twirling-decorator pass can be configured according to various strategies to collect regions of the circuit and surround them by annotated twirling `box`es
+    - Any DD passes can be applied after twirling boxes have been added to ensure that they are compatible with the scheduling constraints implicated by the `box`es. The DD pass can recurse into the boxes, if directed and possible, to perform layer-level DD.
+    - They submit their job. 
+2. The workflow for power users:
+    - Users initialize a `QuantumCircuit` adding boxes manually as they wish.
+    - They submit their job.
+    - Validation routines can be tied to annotations for client-side checking of compatibility with box contents.
+
+Setting up these two workflows presents notable advantages, such as:
+- All of the existing transpiler passes can be asked (at least initially) to simply ignore the boxes, since transpilation is meant to happen before the boxes are generated. 
+- Users of the primitives can inspect the boxes before submitting their jobs. This is a big improvement: today, the box generation happens in the server and can be unintuitive.
+
+In addition to supporting the two workflow above, we believe that the following workflow should *not* be supported:
+
+3. The workflow for "lazy" users:
+    - Users initialize a `QuantumCircuit` without boxes, as they do today.
+    - They apply all of the desired transiler passes, for example to map the circuit to an ISA circuit for the backend that they wish to use.
+    - They submit their job.
+    - The server creates boxes for the users "behind the scenes."
+  
+This workflow essentially takes the same steps as are taken today and therefore presents the same disadvantages discussed above, and also violates the principle "don't do hidden transpilation server-side".
+
+Another application of boxes is to let users declare how they would like to perform noise learning and mitigation. To do so, they can assign a unique identifier to a box, and subsequently use those identifiers in the primitives' options.
```

Comment:
@ihincks FYI

Reply:
we can probably be slightly more specific
```
Another application of boxes is to let users declare how they would like to perform noise learning and mitigation. To do so, they can assign a unique identifier to a box via a separate UID annotation, and subsequently provide the primitives' options with a map from those identifiers to noise models they wish to attach to the identifier.
```